util/gitutil/detectgit.go

```diff
@@ -264,26 +270,37 @@ func gitRelDir(basePath string, path string) (string, bool, error) {
 	if err != nil {
 		return "", false, errors.Wrapf(err, "eval symlinks for %s", absPath)
 	}
-	if !filepath.IsAbs(basePath) {
-		return "", false, errors.Errorf("git base path %s is not absolute", basePath)
-	}
-	basePathSlash := filepath.ToSlash(basePath)
-	pathSlash := filepath.ToSlash(absPath2)
-	basePathParts := strings.Split(basePathSlash, "/")
-	pathParts := strings.Split(pathSlash, "/")
+
+	basePathParts := strings.Split(basePath, string(filepath.Separator))
+	pathParts := strings.Split(absPath2, string(filepath.Separator))
+
+	// `basePath` must be the part of `path`
+	// So it's length split by filepath.Separator must be shorter than `part`.
 	if len(pathParts) < len(basePathParts) {
 		return "", false, nil
 	}
-	for index := range basePathParts {
-		if basePathParts[index] != pathParts[index] {
-			return "", false, nil
-		}
+
+	a, err := os.Stat(filepath.Join(basePathParts...))
+	if err != nil {
+		return "", false, errors.Wrapf(err, "path for %s", basePath)
 	}
-	relPath := strings.Join(pathParts[len(basePathParts):], "/")
+	b, err := os.Stat(filepath.Join(pathParts[len(basePathParts)]))
+	if err != nil {
+		return "", false, errors.Wrapf(err, "path for %s", absPath2)
+	}
+	// Here checks if `path` is included in `basePath` in filesystem agnostic way.
+	// Case-sensitivity difference (like HFS+ in OSX) is also covered by os.SameFile.
+	if !os.SameFile(a, b) {
+		return "", false, errors.New("invalid directories")
+	}
+
+	// Now we are sure that inclusion of `basePath` in `path` is OK.
+	// Finally, here extracts the relative path from `basePath` to return.
+	relPath := filepath.Join(pathParts[len(basePathParts):]...)
 	if relPath == "" {
 		return ".", true, nil
 	}
-	return filepath.FromSlash(relPath), true, nil
```

Comment:
`filepath.Join` internally works like `split.Join(parts, filepath.Separator)`, so now we don't have to call `filepath.FromSlash` manually anymore.